q2_types/feature_data/_format.py

```diff
@@ -100,34 +100,53 @@ class TSVTaxonomyFormat(model.TextFileFormat):
     """
     HEADER = ['Feature ID', 'Taxon']
 
-    def sniff(self):
+    def _check_n_records(self, n=None):
         with self.open() as fh:
             data_lines = 0
             header = None
-            while data_lines < 10:
-                line = fh.readline()
+
+            file_ = enumerate(fh) if n is None else zip(range(n), fh)
+
+            for i, line in file_:
+                i = i + 1
 
                 if line == '':
                     # EOF
                     break
-                elif line.lstrip(' ') == '\n':
-                    # Blank line
+                elif line.strip(' ') == '\n':
                     continue
                 elif line.startswith('#'):
                     # Comment line
                     continue
 
-                cells = line.rstrip('\n').split('\t')
+                cells = line.strip('\n').split('\t')
+
                 if header is None:
                     if cells[:2] != self.HEADER:
-                        return False
+                        raise ValidationError("['Feature ID' and 'Taxon'] "
+                                              "must be the first two header "
+                                              "values to be a valid taxonomy "
+                                              "file.\n\nThe first two header "
+                                              "values provided are: {}."
+                                              .format(cells[:2]))
                     header = cells
                 else:
                     if len(cells) != len(header):
-                        return False
+                        raise ValidationError('Number of headers are not the '
```

Comment:
Please flip the order of the sentence --- I think it would be more clear if it said that the number of column values (not columns), doesn't match the number of header values. I like the last bit that shows the header values and the column values. :cactus:

Reply:
I flipped the order of the sentence in this error message.